services/payments/service.go

```diff
@@ -164,7 +164,7 @@ func (s *Service) configureKMSEncryptionKey(ctx context.Context) error {
 			return nil
 		}
 		logger.Info().Msgf("policy does not match: \n\n %s \n\n %s!", *getKeyPolicyResult.Policy, policy)
-		return fmt.Errorf("failed to match policy text")
+		panic("failed to match policy text")
```

Comment:
I'd argue that a better way would have been to return a sentinel error, propagate it up, and handle it where appropriate. This happens in an initialising code, and this code is called in `NewService` which already returns an error.

What does stop us from doing that?

Reply:
I agree but this is causing active problems during payouts. Making an issue off your comment to address properly.